docs/upgrading.md

```diff
@@ -0,0 +1,62 @@
+# Upgrading
```

Comment:
+1. Breaking out this guidance about major version upgrades from the bulk of `CHANGELOG` is a great idea. It really emphasizes what there is to know about this upgrading (or if somehow we've overlooked something, what the gaps are). Big +1 from me, nicely done.